File: ServeurGeo2D/src/Construction/ConstructeurGroupe.java
package Construction;

import java.awt.Graphics;

import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import Erreur.Erreur;

public class ConstructeurGroupe extends ConstructeurCOR
{

	ConstructeurGroupe(ConstructeurCOR constructeurSuivant) {
		super(constructeurSuivant);
		
	}

	public boolean resoudre1(Node s, Graphics g) 
	{
		if(s.getNodeName().equalsIgnoreCase("groupe"))
		{
			ConstructeurCOR COR = new ConstructeurCercle(null);
			COR = new ConstructeurSegment(COR);
			COR = new ConstructeurTriangle(COR);
			COR = new ConstructeurPolygone(COR);
			COR = new ConstructeurGroupe(COR);
			NodeList racineNoeuds = s.getChildNodes();
		    int nbRacineNoeuds = racineNoeuds.getLength();
		    for (int i = 0; i<nbRacineNoeuds; i++) 
		    {
		    	if(racineNoeuds.item(i).getNodeType() == Node.ELEMENT_NODE) 
		    	{
		    	        final Node figure = racineNoeuds.item(i);
		    	        if(COR.resoudre(figure,g) == false)
		    	        	new Erreur("Erreur de construction, figure inconnu.");
		    	}
		    }
			return true;
		}
		else
		{
			return false;
		}
	}
}

File: Client_Simple/Constructeur_Sources/ConstructeurGroupe.cpp
#include "../Constructeur_Headers/ConstructeurGroupe.h"

ConstructeurGroupe::ConstructeurGroupe(ConstructeurCOR *s) : ConstructeurCOR(s)
{

}

Figure *ConstructeurGroupe::resoudre1(const QDomNode *d) const
{
    QDomNode *noeud = new QDomNode(d->firstChild());

    //Appel des constructeurs
    ConstructeurCOR *COR = new ConstructeurCercle(NULL);
    COR = new ConstructeurSegment(COR);
    COR = new ConstructeurPolygone(COR);
    COR = new ConstructeurTriangle(COR);
    COR = new ConstructeurGroupe(COR);

    Groupe *g = new Groupe(Couleur::BLACK);
    while(!noeud->isNull())
    {
            //On envoie le noeud à l'expert
            Figure *f = COR->resoudre(noeud);
            if(f == NULL)
            {
                cout << "Erreur figure inconnu, fichier corrompu." << endl;
                return NULL;
            }
            else
            {
                g->add(f);
                *noeud = noeud->nextSibling();//On passe à l'objet suivant
            }
    }
    return g;
}
